src/AgentVisibilityContext.tsx

```diff
@@ -55,8 +63,21 @@ export const VisibilityProvider = ({
     }, {});
   }, [uiDisplayData]);
 
+  /**
+   * This is never used as an argument but should be set upon arrival
+   * of the uiDisplayData to ensure that the highlightedAgents state
+   * is not an empty object, but corresponds to the actual agent data.
+   */
+  const noAgentsHighlighted: VisibilitySelectionMap = useMemo(() => {
+    return uiDisplayData.reduce<VisibilitySelectionMap>((acc, item) => {
+      acc[item.name] = [item.name];
+      return acc;
+    }, {});
+  }, [uiDisplayData]);
```

Comment:
can't you use the "onUiDisplayDataArrive" or whatever it's called? Might be slightly less confusing than this setup

Reply:
Sure I could set them in that handler. I just need to use state variables to hold the initial values I think.